ci/before_install.sh

```diff
@@ -7,28 +7,34 @@ if [[ "${CI_TARGET}" == lint ]]; then
   exit
 fi
 
-echo 'python info:'
+echo 'Python info:'
 (
-  2>&1 python --version || true
-  2>&1 python2 --version || true
-  2>&1 python3 --version || true
-  2>&1 pip --version || true
-  2>&1 pip2 --version || true
-  2>&1 pip3 --version || true
-  echo 'pyenv versions:'
-  2>&1 pyenv versions || true
-) | sed 's/^/  /'
-
-if [[ "${TRAVIS_OS_NAME}" == osx ]]; then
-  echo "Upgrade Python 3 pip"
-  python3 -m pip -q install --user --upgrade pip
-else
-  echo "Upgrade Python 2 pip"
-  python2.7 -m pip -q install --user --upgrade pip
-  echo "Upgrade Python 3 pip"
-  # Allow failure. pyenv pip3 on travis is broken:
-  # https://github.com/travis-ci/travis-ci/issues/8363
-  python3 -m pip -q install --user --upgrade pip || true
+  set -x
+  python3 --version
+  python2 --version
+  python --version
+  pip3 --version
+  pip2 --version
+  pip --version
+  pyenv versions
+) 2>&1 | sed 's/^/  /' || true
+
+# Use pyenv, but not for OSX on Travis, where it only has the "system" version.
+if [[ "${TRAVIS_OS_NAME}" != osx ]] && command -v pyenv; then
+  echo 'Setting Python versions via pyenv'
+  # Prefer python2 as python for /usr/bin/asan_symbolize-4.0.
+  pyenv global 2.7.15:3.7
```

Comment:
using pyenv was already tried before: https://github.com/neovim/neovim/pulls?q=is%3Apr+pyenv+is%3Aclosed

It doesn't really fix the problems with python on travis. Are you sure it works now?

Reply:
> I really need to look at these PRs before they are merged.

I see, but I've waited several days and figured it would be good to move on.

(this was related to progressing on Python/pynvim setup in general)

I will be more passive here in the future I guess.

Reply:
As for the pyenv setup itself: it is more straight forward now in my opinion.

Reply:
More straightfoward how? `python -m pip` is the python-recommended way of invoking `pip`. Using `pip` directly is deprecated, and it is imprecise because different versions of python may exist on a system.

Reply:
> I will be more passive here in the future I guess.

It's good to be active but when reverting previous change we usually try to get signoff by the previous author. Otherwise it causes circular changes.

Reply:
> More straightfoward how? 

It explicitly uses it, and hsa python3/pip3 now installed always.

Reply:
Check `git log --grep pyenv` , we tried this in the past and it didn't work, because of an unresolved Travis issue.